tests/test_kv_manipulation.py

```diff
@@ -4,14 +4,10 @@
     import common as c
 
 
-class TestSplinepyKnotVectorManipulation(c.unittest.TestCase):
+class TestSplinepyKnotVectorManipulation(c.SplineBasedTestCase):
     def setUp(self):
-        self.b2P2D = c.b2P2D.copy()
-        self.n2P2D = c.n2P2D.copy()
-        self.bspline = c.splinepy.BSpline(**self.b2P2D)
-        self.nurbs = c.splinepy.NURBS(**self.n2P2D)
-        self.ref_bspline = c.splinepy.BSpline(**c.b2P2D)
-        self.ref_nurbs = c.splinepy.NURBS(**c.n2P2D)
+        self.bspline = self.spline_b2p2d()
+        self.nurbs = self.spline_n2p2d()
```

Comment:
.

Reply:
Here the knot vectors of the splines are edited, so the same spline must be used.